pyodide_lock/spec.py

```diff
@@ -91,3 +101,151 @@ def check_wheel_filenames(self) -> None:
                 for name, errs in errors.items()
             )
             raise ValueError(error_msg)
+
+    def add_wheels(  # noqa: C901
+        self,
+        wheel_files: list[Path],
+        base_path: Path | None = None,
+        base_url: str = "",
+    ) -> None:
+        """Add a list of wheel files to this pyodide-lock.json
+
+        Args:
+            wheel_files (list[Path]): A list of wheel files to import.
+            base_path (Path | None, optional):
+                Filenames are stored relative to this base path. By default the
+                filename is stored relative to the path of the first wheel file
+                in the list.
+
+            base_url (str, optional):
+                The base URL stored in the pyodide-lock.json. By default this
+                is empty which means that wheels must be stored in the same folder
+                as the core pyodide packages you are using. If you want to store
+                your custom wheels somewhere else, set this base_url to point to it.
+        """
+        if len(wheel_files) <= 0:
+            return
+        if base_path is None:
+            base_path = wheel_files[0].parent
+
+        target_python = version_parse(self.info.python)
+        python_binary_tag = f"cp{target_python.major}{target_python.minor}"
+        python_pure_tags = [
+            f"py2.py{target_python.major}",
+            f"py{target_python.major}",
+            f"py{target_python.major}{target_python.minor}",
+        ]
+
+        target_platform = self.info.platform + "_" + self.info.arch
+
+        new_packages = {}
+        new_package_wheels = {}
+        for f in wheel_files:
+            split_name = f.stem.split("-")
+            name = canonicalize_name(split_name[0])
+            version = split_name[1]
+            python_tag = split_name[-3]
+            split_name[-2]
+            platform_tag = split_name[-1]
```

Comment:
This can be done more reliably with https://packaging.pypa.io/en/stable/utils.html#packaging.utils.parse_wheel_filename